derive/examples/meta-macros.rs

```diff
@@ -36,6 +36,10 @@ pub trait Rpc<One> {
 	/// Performs an asynchronous operation with meta.
 	#[rpc(meta, name = "callAsyncMeta", alias("callAsyncMetaAlias"))]
 	fn call_meta(&self, a: Self::Metadata, b: BTreeMap<String, Value>) -> FutureResult<String, Error>;
+
+	/// Responds to a notification.
+	#[notification(name = "notify")]
```

Comment:
Do we actually need a different attribute? Can't we detect that the return type is `()` and act upon that?ac

Reply:
I think that works. We can certainly do that!

Reply:
The main reason why I chose a different attribute is so we can reject the `returns` key from being used with notifications, but I think we could probably make that work. Maybe we could emit an error for functions annotated with `returns` that also return a `()`?

Reply:
Okay, I believe this is doable! We could remove the `AttributeKind::Notification` variant and add an `is_notification: bool` field to `AttributeKind::Rpc`.

This is necessary because without some kind of marker as to whether an `rpc` method is a true method or a notification, we would need to check the return type twice, both in `rpc_trait.rs` and `to_delegate.rs`. Doing so just once in `parse_meta()` inside `rpc_attr.rs` when we first determine the `AttributeKind` is the most efficient way to do this, I think.

We can also use this opportunity to make sure that the `returns` meta word isn't being used on notifications.

Reply:
Okay, I believe your suggestion is doable! We could remove the `AttributeKind::Notification` variant and add an `is_notification: bool` field to `AttributeKind::Rpc`.

This is necessary because without some kind of marker as to whether an `rpc` method is a true method or a notification, we would need to check the return type twice, both in `rpc_trait.rs` and `to_delegate.rs`. Doing so just once in `parse_meta()` inside `rpc_attr.rs` when we first determine the `AttributeKind` is the most efficient way to do this, I think.

We can also use this opportunity to make sure that the `returns` meta word isn't being used on notifications.

Reply:
Done in https://github.com/paritytech/jsonrpc/pull/454/commits/f4f20fc07f5c451f2c956c04af6fbce0b91e8e3f.